code/renderers/vue3/src/render.ts

```diff
@@ -49,12 +81,33 @@ export function renderToDOM(
       Use "() => ({ template: '<my-comp></my-comp>' })" or "() => ({ components: MyComp, template: '<my-comp></my-comp>' })" when defining the story.
       `,
     });
-    return;
+    return () => {
+      teardown(domElementKey);
+    };
   }
 
-  showMain();
+  if (!existingApp || forceRemount) {
+    const reactiveArgs = args ? reactive(args) : args;
 
-  map.get(domElement)?.unmount();
+    // TODO: explain cyclical nature of these app => story => mount
+    const app = createApp({
+      render() {
+        appsAndArgsByDomElementKey.set(domElementKey, { app, reactiveArgs });
+        return h(element, reactiveArgs);
+      },
+    });
+    app.config.errorHandler = (e: unknown) => showException(e as Error);
+    setupFunction(app);
```

Comment:
Previously `setupFunction(app)` was invoked from within the `render()` function, but that is an issue if we keep the component, because it will mean that globals will be set on every render.

The assumption was that `setupFunction()` doesn't need to be called from within `render`, is that correct?